### PR TITLE
Fix early return preventing mutex lock in attempt_restart_check

### PR DESCRIPTION
**Pull request by guilhermelawless**
Best reviewed without whitespace changes.
Thanks to sev for the detailed report.

### CODE REVIEW

File: nano/node/bootstrap/bootstrap.cpp [hunk 1]
```diff
@@ -617,6 +617,7 @@ void nano::bootstrap_attempt::attempt_restart_check (nano::unique_lock<std::mute
 	if (!frontiers_confirmed && (requeued_pulls > (!node->network_params.network.is_test_network () ? nano::bootstrap_limits::requeued_pulls_limit : nano::bootstrap_limits::requeued_pulls_limit_test) || total_blocks > nano::bootstrap_limits::frontier_confirmation_blocks_limit))
 	{
 		confirm_frontiers (lock_a);
+		assert (!lock_a.try_lock ());
```
Comment (wezrule) [inline nano/node/bootstrap/bootstrap.cpp]: "If try_lock is called by a thread that already owns the mutex, the behavior is undefined." https://en.cppreference.com/w/cpp/thread/mutex/try_lock
Can that happen here?

Comment (guilhermelawless) [reply nano/node/bootstrap/bootstrap.cpp]: Interesting. No such mention in https://en.cppreference.com/w/cpp/thread/unique_lock/try_lock (for unique locks), although we use this idiom at least 10 times throughout the code directly with mutexes.

Comment (wezrule) [reply nano/node/bootstrap/bootstrap.cpp]: It also says it effectively just calls `mutex.try_lock`. I guess they are all surrounded in `assert`s so no user would see anything anyway, so it's probably fine.

Comment (guilhermelawless) [reply nano/node/bootstrap/bootstrap.cpp]: Do you think it's better to move it outside the assert and use `owns_lock()` ?

Comment (wezrule) [reply nano/node/bootstrap/bootstrap.cpp]: `owns_lock` is probably better yes

**Pull request merged**